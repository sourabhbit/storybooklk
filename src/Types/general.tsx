export enum ComponentSizeENUM {
	small = 'small',
	medium = 'medium',
	large = 'large',
}

export enum TypographyENUM {
	sans = 'var(--font-sans)',
	serif = 'var(--font-serif)',
}

export enum ThemeENUM {
	primary = 'primary',
	secondary = 'secondary',
}
