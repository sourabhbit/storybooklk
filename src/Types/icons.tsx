export enum iconsENUM {
	addToCart = 'addToCart',
	addedToCart = 'addedToCart',
	antiScratch = 'antiScratch',
	backArrow = 'backArrow',
	bin = 'bin',
	cross = 'cross',
	crossWithSpace = 'crossWithSpace',
	edit = 'edit',
	emptyCart = 'emptyCart',
	fullWidth = 'fullWidth',
	gift = 'gift',
	glasses = 'glasses',
	globe = 'globe',
	gridList = 'gridList',
	phone = 'phone',
	share = 'share',
	star = 'star',
	starFilled = 'starFilled',
	tick = 'tick',
	user = 'user',
	group = 'group',
}
