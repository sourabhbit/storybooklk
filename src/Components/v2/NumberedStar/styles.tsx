import styled from 'styled-components'
import { ComponentSizeENUM, TypographyENUM } from 'Types/general'

const getFontSize = (size: ComponentSizeENUM) => {
	if (size === ComponentSizeENUM.small) {
		return 'var(--fs-12)'
	} else if (size === ComponentSizeENUM.medium) {
		return 'var(--fs-18)'
	} else {
		return 'var(--fs-30)'
	}
}
export const NumberedRatingWrapper = styled.div<{
	styleSize: ComponentSizeENUM
}>`
	display: flex;
	align-items: center;
`

export const NumberedRatingSpan = styled.span<{
	styleSize: ComponentSizeENUM
	styleFont: TypographyENUM
}>`
	color: var(--dark-blue-100);
	margin-left: var(--spacing-sm);
	font-size: ${(props) => getFontSize(props.styleSize)};
	font-family: ${(props) => props.styleFont};
	font-weight: var(--fw-regular);
`
