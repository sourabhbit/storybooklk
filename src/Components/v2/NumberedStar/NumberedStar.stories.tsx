import { ComponentMeta, ComponentStory } from '@storybook/react'
import { generateArgTypes } from 'Helpers/Storybook'
import { ComponentSizeENUM, TypographyENUM } from 'Types/general'
import { iconsENUM } from 'Types/icons'
import NumberedStar from './index'

export default {
	title: 'Components/v2/Rating/NumberedStar',
	component: NumberedStar,
	argTypes: {
		id: generateArgTypes({
			type: 'string',
			required: true,
			description: 'String used as unique identifier',
			category: 'General',
		}),
		componentSize: generateArgTypes({
			type: 'string',
			required: false,
			description: 'Size of the component',
			category: 'General',
			defaultValue: ComponentSizeENUM.small,
			control: 'radio',
			options: ComponentSizeENUM,
		}),
		font: generateArgTypes({
			type: 'string',
			required: true,
			description: 'Typography of the component',
			category: 'General',
			defaultValue: TypographyENUM.serif,
			control: 'radio',
			options: TypographyENUM,
		}),
		rating: generateArgTypes({
			type: 'number',
			required: true,
			description: 'Rating count',
			category: 'Component',
		}),
		icon: generateArgTypes({
			type: 'string',
			required: false,
			description: 'Rating Icon',
			category: 'Component',
			control: 'select',
			defaultValue: 'starFilled',
			options: iconsENUM,
		}),
	},
} as ComponentMeta<typeof NumberedStar>

const Template: ComponentStory<typeof NumberedStar> = (args) => (
	<NumberedStar {...args} />
)

export const Overview = Template.bind({})
Overview.args = {
	id: 'numbered-star',
	rating: 4.5,
	componentSize: ComponentSizeENUM.medium,
	icon: iconsENUM.starFilled,
	font: TypographyENUM.serif,
}
