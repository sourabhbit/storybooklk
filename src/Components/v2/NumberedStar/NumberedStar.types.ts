import { ComponentSizeENUM, TypographyENUM } from 'Types/general'
import { iconsENUM } from 'Types/icons'
export interface NumberedStarTypes {
	componentSize?: ComponentSizeENUM
	rating: number
	id: string
	font: TypographyENUM
	icon?: iconsENUM
}
