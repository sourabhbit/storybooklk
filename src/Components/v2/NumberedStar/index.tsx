import { NumberedStarTypes } from './NumberedStar.types'
import { NumberedRatingWrapper, NumberedRatingSpan } from './styles'
import { ComponentSizeENUM, TypographyENUM } from 'Types/general'
import SVGIcon from 'Components/common/SVGIcon'
const NumberedStar = ({
	componentSize = ComponentSizeENUM.medium,
	rating = 4.5,
	id,
	font = TypographyENUM.serif,
	icon,
}: NumberedStarTypes) => {
	return (
		<NumberedRatingWrapper
			styleSize={componentSize}
			data-testid="numbered-test"
			id={id}
		>
			{icon && <SVGIcon icon={icon} alt="star" />}
			<NumberedRatingSpan styleSize={componentSize} styleFont={font}>
				{rating}
			</NumberedRatingSpan>
		</NumberedRatingWrapper>
	)
}

export default NumberedStar
