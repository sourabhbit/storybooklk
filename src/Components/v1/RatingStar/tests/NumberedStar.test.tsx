import { render, screen } from '@testing-library/react'
import { TypographyENUM } from 'Types/general'
import NumberedStar from '../index'

const props = {
	rating: 4.5,
	id: 'numberedStar',
	font: TypographyENUM.sans,
}

describe('<NumberedStar />', () => {
	let wrapper: any = null

	beforeEach(() => {
		const { asFragment } = render(<NumberedStar {...props} />)
		wrapper = asFragment
	})

	afterAll(() => {
		wrapper = null
	})

	it('Should take a snapshot', () => {
		expect(wrapper()).toMatchSnapshot()
	})
	it('Should render Numbered Star Component', () => {
		const container = screen.getByTestId('numbered-test')
		expect(container).toBeInTheDocument()
	})
})
