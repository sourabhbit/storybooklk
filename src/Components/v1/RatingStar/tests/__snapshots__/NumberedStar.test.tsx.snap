// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberedStar /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL dYfRsX"
    data-testid="numbered-test"
  >
    <span
      class="sc-gsDKAQ bnvAvb"
    >
      4.5
    </span>
  </div>
</DocumentFragment>
`;
